tests/data/configs/validate_conf.py

```diff
@@ -0,0 +1,192 @@
+from src.modules.configs.config import generate_string
+
+application = {
+    "unrecognized_key": {
+        "container": {
+            "sata": {
+                "device_path": "/dev/sda",
+                "mount_path": "/mnt"
+            }
+        }
+    },
+    "mandatory_key1": {
+        "containers": {
+            "sata": {
+                "device_path": "/dev/sda"
+            }
+        }
+    },
+    "mandatory_key2": {
+        "resources": {
+            "thread_count": 1
+        },
+        "workdir": "/var/log/ingestation"
+    },
+    "non_mandatory_key": {
+        "containers": {
+            "sata": {
+                "device_path": "/dev/sda",
+                "mount_path": "/mnt"
+            }
+        },
+        "exclude": [
+            "EX_DISK_APP"
+        ]
+    },
+    "invalid_data_type": {
+        "containers": {
+            "sata": {
+                "device_path": 1,
+                "mount_path": "/mnt"
+            }
+        }
+    },
+    "too_long_value": {
+        "containers": {
+            "sata": {
+                "device_path": generate_string(4097),
+                "mount_path": "/mnt"
+            }
+        }
+    },
+    "successful_validation": {
```

Comment:
Imo successful case should be separated from failed one, both in test data and in code - few more lines but much clearer picture.